src/Credentials/ServiceAccountCredentials.php

```diff
@@ -219,11 +219,20 @@ public function fetchAuthToken(callable $httpHandler = null)
     }
 
     /**
+     * Return the Cache Key for the credentials.
+     * For the cache key format is:
+     * ClientEmail:Scope | Audience:sub.
+     *
      * @return string
      */
     public function getCacheKey()
     {
-        $key = $this->auth->getIssuer() . ':' . $this->auth->getCacheKey();
+        $scopeOrAudience = $this->auth->getScope();
+        if (!$scopeOrAudience) {
+            $scopeOrAudience = $this->auth->getAudience();
+        }
+
+        $key = $this->auth->getIssuer() . ':' . $scopeOrAudience;
```

Comment:
Let's remove the use of colon, or replace with a `.`, since we would strip it later anyway. 
```suggestion
        $key = $this->auth->getIssuer() . '.' . $scopeOrAudience;
```

(and do this below as well)